Podcast/SeriesDetailViewController.swift

```diff
@@ -20,76 +21,139 @@ class SeriesDetailViewController: UIViewController, SeriesDetailHeaderViewDelega
+        var seriesHeaderViewY: CGFloat = 0
+        if let height = navigationController?.navigationBar.frame.maxY  {
+            seriesHeaderViewY = height
+        }
+        epsiodeTableView = UITableView(frame:  CGRect(x: 0, y: seriesHeaderViewY, width: view.frame.width, height: view.frame.height - seriesHeaderViewY))
         epsiodeTableView.delegate = self
         epsiodeTableView.dataSource = self
         epsiodeTableView.tableHeaderView = seriesHeaderView
         epsiodeTableView.showsVerticalScrollIndicator = false
+        epsiodeTableView.separatorStyle = .none
+        epsiodeTableView.contentInset = UIEdgeInsetsMake(0, 0, appDelegate.tabBarController.tabBarHeight, 0)
+        epsiodeTableView.addInfiniteScroll { (tableView) -> Void in
+            self.fetchEpisodes()
+        }
+        //tells the infinite scroll when to stop
+        epsiodeTableView.setShouldShowInfiniteScrollHandler { _ -> Bool in
+            return self.continueInfiniteScroll
+        }
         epsiodeTableView.register(EpisodeTableViewCell.self, forCellReuseIdentifier: "EpisodeTableViewCellIdentifier")
-        
         view.addSubview(epsiodeTableView)
+    
+        epsiodeTableView.infiniteScrollIndicatorView = createLoadingAnimationView()
+        
+        if let series = self.series {
+            seriesHeaderView.setSeries(series: series)
+            navigationController?.title = series.title
+            fetchEpisodes()
+        }
+        
+        automaticallyAdjustsScrollViewInsets = false
+        
+        loadingAnimation = createLoadingAnimationView()
+        loadingAnimation.center = seriesHeaderView.center
+        view.addSubview(loadingAnimation)
+        loadingAnimation.startAnimating()
     }
     
-    func fetchSeries() {
-        // For now dummy data, later endpoint request
-
-        // Setup dummy data
-        let s = Series()
-        s.title = "Dog Pods"
-        s.descriptionText = "We talk lots about dogs and puppies and how cute they are and the different colors they come in and how fun they are."
-        s.author = "Dog Lovers"
-        s.tags = [Tag(name: "Design")]
-        let episode = Episode()
-        episode.title = "Puppies Galore"
-        episode.series = s
-        episode.dateCreated = Date()
-        episode.descriptionText = "We talk lots about dogs and puppies and how cute they are and the different colors they come in and how fun they are."
-        episode.tags = [Tag(name:"Design"), Tag(name:"Learning"), Tag(name: "User Experience"), Tag(name:"Technology"), Tag(name:"Innovation"), Tag(name:"Dogs")]
-        s.episodes = [episode]
-        series = s
-    }
-    
-    func updateViewsWithSeries(series: Series) {
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        if self.series != nil {
+            //load animation for awhile for better UX
+            let deadlineTime = DispatchTime.now() + .milliseconds(100)
```

Comment:
oh okay so i thought UX perspective it looked a little weird to have the series just show up .. so now what happens is the loader animates for a second then series header view shows up. It looks better this way